helpers/scrolling.ts

```diff
@@ -139,18 +145,52 @@ export function scrollHorizontally(
   if (left - parentLeft < opts.leftOffset) {
     const goal = scroll.x - opts.leftOffset + left - parentLeft;
     if (opts.smooth) {
-      smoothScrollLeft(goal, parent);
+      addToScrollQueue(smoothScrollLeft, parent, goal);
     } else {
-      scrollElement(parent, goal, scroll.y);
+      addToScrollQueue(scrollElement, parent, goal, scroll.y);
     }
   }
 
   if (right - parentLeft > width - opts.rightOffset) {
     const goal = scroll.x + right - parentLeft - width + opts.rightOffset;
     if (opts.smooth) {
-      smoothScrollLeft(goal, parent);
+      addToScrollQueue(smoothScrollLeft, parent, goal);
     } else {
-      scrollElement(parent, goal, scroll.y);
+      addToScrollQueue(scrollElement, parent, goal, scroll.y);
     }
   }
 }
+
+type ScrollFunction = (element: Element | Window, x: number, y?: number) => Promise<void>;
+
+type ScrollQueueItem = {
+  element: Element | Window;
+  fn: ScrollFunction;
+  x: number;
+  y?: number;
+};
+
+const queue: ScrollQueueItem[] = [];
+let queueRunning: boolean;
+
+function addToScrollQueue(
+  fn: ScrollFunction,
+  element: Element | Window,
+  x: number,
+  y?: number
+): void {
+  queue.push({ fn, element, x, y });
+  if (!queueRunning) {
+    processQueue();
+  }
+}
+
+function processQueue() {
+  if (queue.length > 0) {
+    queueRunning = true;
+    const { fn, element, x, y } = queue.splice(0, 1)[0];
```

Comment:
Maybe use `shift` here instead of `splice`, because that's in fact what you are doing.

Reply:
Of course 👍 !